packages/public-api/src/clientApiRouter.ts

```diff
@@ -49,6 +50,43 @@ type RequestWebPageScanResponse = {
 
 export const appRouter = trpc
   .router<Context>()
+  .query('getShowcase', {
+    async resolve() {
+      const showcaseData = await getShowcaseData();
+
+      const showcasedScans = showcaseData.showcasedScans.map((showcasedScan) => ({
+        hostname: {
+          hostname: showcasedScan.scan.webPage.hostname.hostname,
+        },
+        webPage: {
+          path: showcasedScan.scan.webPage.path,
+        },
+        scanPreview: {
+          packageNames:
+            showcasedScan.scan.scanResult?.identifiedPackages.slice(0, 6).map((pkg) => pkg.name) ??
+            [],
+          totalCount: showcasedScan.scan.scanResult?.identifiedPackages.length ?? 0,
+        },
+      }));
+
+      const scansWithVulnerabilities = showcaseData.scansWithVulnerabilities.map(
+        (scanWithVulnerabilities) => ({
+          hostname: {
+            hostname: scanWithVulnerabilities.sourceScan.webPage.hostname.hostname,
```

Comment:
Why such weird nesting?

Reply:
These are, effectively, entities. I expect to land additional fields like `favicon` here later. Anyways, this is non-persistent data structure and can be easily refactored later.